apps/vaults/hooks/useSolverCowswap.tsx

```diff
@@ -0,0 +1,300 @@
+
+		const	{quote} = latestQuote.current;
+		try {
+			const	buyAmountWithSlippage = getBuyAmountWithSlippage(latestQuote.current, request.current.outputToken.decimals);
+			const	_signature = await signCowswapOrder({
+				...quote,
+				buyAmount: buyAmountWithSlippage.toString()
+			});
+			signature.current = _signature;
+			return true;
+		} catch (_error) {
+			console.error(_error);
+			return false;
+		}
+	}, [latestQuote, signCowswapOrder]);
+
+	/* 🔵 - Yearn Finance **************************************************************************
+	** Cowswap orders have a validity period and the return value on submit is not the execution
+	** status of the order. This method is used to check the status of the order and returns a
+	** boolean value indicating whether the order was successful or not.
+	** It will timeout once the order is no longer valid or after 50 minutes (max should be 30mn)
+	**********************************************************************************************/
+	async function checkOrderStatus(orderUID: string, validTo: Timestamp): Promise<{isSuccessful: boolean, error: Error | undefined}> {
+		for (let i = 0; i < maxIterations; i++) {
+			const {data: order} = await axios.get(`https://api.cow.fi/mainnet/api/v1/orders/${orderUID}`);
+			if (order?.status === 'fulfilled') {
+				return ({isSuccessful: true, error: undefined});
+			} else if (order?.status === 'cancelled' || order?.status === 'expired') {
+				return ({isSuccessful: false, error: new Error('TX fail because the order was not fulfilled')});
+			}
+			if (validTo < (new Date().valueOf() / 1000)) {
+				return ({isSuccessful: false, error: new Error('TX fail because the order expired')});
+			}
+			// Sleep for 3 seconds before checking the status again
+			await new Promise((resolve): NodeJS.Timeout => setTimeout(resolve, 3000));
+		}
+		return ({isSuccessful: false, error: new Error('TX fail because the order expired')});
+	} 
+
+	/* 🔵 - Yearn Finance **************************************************************************
+	** execute will send the post request to execute the order and wait for it to be executed, no
+	** matter the result. It returns a boolean value indicating whether the order was successful or
+	** not.
+	**********************************************************************************************/
+	const execute = useCallback(async (): Promise<boolean> => {
+		if (!latestQuote?.current || !latestQuote?.current?.quote || !request.current) {
+			return false;
+		}
+		const	{quote, from, id} = latestQuote.current;
+		try {
+			const	buyAmountWithSlippage = getBuyAmountWithSlippage(latestQuote.current, request.current.outputToken.decimals);
+			const	{data: orderUID} = await axios.post('https://api.cow.fi/mainnet/api/v1/orders', {
+				...quote,
+				buyAmount: buyAmountWithSlippage.toString(),
+				from: from,
+				quoteId: id,
+				signature: signature.current,
+				signingScheme: String(shouldUsePresign ? 'presign' : 'eip712')
```

Comment:
I doubt typescript would not be able to infer these types 
```suggestion
				signingScheme: shouldUsePresign ? 'presign' : 'eip712'
```